public/app/plugins/panel/geomap/GeomapPanel.tsx

```diff
@@ -505,6 +505,9 @@ export class GeomapPanel extends Component<Props, State> {
 
     const handler = await item.create(map, options, config.theme2);
     const layer = handler.init();
+    if (options.opacity != null) {
```

Comment:
maybe also:
```
&& !layer.hideOpacity
```

Reply:
options.opacity is undefined when hideOpacity is true, so we should be fine here.